tel2puml/__main__.py

```diff
@@ -18,67 +18,238 @@
-    default="./default.puml",
-    dest="puml_file_path",
+pv_exclusive_group.add_argument(
+    "file_paths",
+    nargs="*",
+    help="Input .json files containing job data",
+    default=[],
 )
 
-parser.add_argument(
-    "-sn",
-    "--sequence-name",
+pv_to_puml_parser.add_argument(
+    "-jn",
+    "--job-name",
     metavar="name",
-    help="Name given to the puml sequence diagram",
+    help=(
+        "Name given to the puml sequence diagram and prefix for the output "
+        "puml file"
+    ),
     default="default_name",
-    dest="puml_name",
-)
-
-parser.add_argument(
-    "file_paths",
-    nargs="*",
-    help="Input .json files containing job data",
+    dest="job_name",
 )
 
-parser.add_argument(
+pv_to_puml_parser.add_argument(
     "-group-by-job",
     help="Group events by job ID",
     action="store_true",
     dest="group_by_job",
 )
 
 
-if __name__ == "__main__":
-    args: argparse.Namespace = parser.parse_args()
+def generate_config(file_path: str) -> Any:
+    """Parse config file.
+
+    :param file_path: The filepath to the config file
+    :type file_path: `str`
+    :return: Config file represented as a dictionary
+    :rtype: `dict`[`str`, `Any`]
+    """
+    with open(file_path, "r") as file:
+        return yaml.safe_load(file)
+
+
+def validate_inputs(args: argparse.Namespace) -> dict[str, Any]:
+    """Validate CLI inputs using pydantic models.
+
+    :param args: argparse object
+    :type args: :class:`argparse.Namespace`
+    :return: CLI arguments as a dictionary
+    :rtype: `dict`[`str`, `Any`]
+    """
+
     args_dict = vars(args)
-    if args_dict["file_paths"]:
-        args_dict.pop("folder_path")
-        args_dict.pop("group_by_job")
-        pv_jobs_from_files_to_puml_file(**args_dict)
-    elif args_dict["group_by_job"]:
-        args_dict.pop("file_paths")
-        pv_events_from_folder_to_puml_file(**args_dict)
+    if args.command == "otel2puml":
+        OtelToPumlArgs(**args_dict)
+    elif args.command == "otel2pv":
+        OtelToPvArgs(**args_dict)
+    elif args.command == "pv2puml":
+        PvToPumlArgs(**args_dict)
     else:
-        args_dict.pop("file_paths")
-        args_dict.pop("group_by_job")
-        pv_jobs_from_folder_to_puml_file(**args_dict)
+        raise ValueError("No subcommand selected.")
+
+    return args_dict
+
+
+def find_json_files(directory: str) -> list[str]:
+    """Walk a directory path and extract .json files.
+
+    :param directory: The directory path
+    :type directory: `str`
+    :return: List of filepaths
+    :rtype: `list`[`str`]
+    """
+    json_files = []
+
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".json"):
+                json_files.append(os.path.join(root, file))
+
+    if not json_files:
+        raise FileNotFoundError(
+            f"No .json files were found in directory '{directory}'"
+        )
+
+    return json_files
+
+
+def generate_puml_options(
```

Comment:
I think this is fine arguments wise but I think we should do validation in this function as types will be more specific and less open ended.